### PR TITLE
[core] Manifest and Statistic support query by tag name

### PR DESCRIPTION
**Pull request by xuzifu666**
### Purpose

Linked issue: close #xxx
Follow up https://github.com/apache/paimon/pull/1667 and https://github.com/apache/paimon/pull/4251 
which can only support query by snapshot id, this pr is aimed to support tag name query.

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java [hunk 1]
```diff
@@ -170,8 +171,15 @@ public Identifier identifier() {
     public Optional<Statistics> statistics() {
         Snapshot latestSnapshot;
         Long snapshotId = coreOptions().scanSnapshotId();
+        String tagName = coreOptions().scanTagName();
+
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java:175]: Here add a check, if both snapshotId and tagName are not null, then there is ambiguity here.

File: paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java [hunk 2]
```diff
@@ -211,7 +213,11 @@ private static List<ManifestFileMeta> allManifests(FileStoreTable dataTable) {
             }
             snapshot = snapshotManager.snapshot(snapshotId);
         } else if (snapshotId == null) {
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java:215]: here, Condition 'snapshotId == null' is always 'true'

File: paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java [hunk 3]
```diff
@@ -198,6 +199,7 @@ private static List<ManifestFileMeta> allManifests(FileStoreTable dataTable) {
         CoreOptions coreOptions = CoreOptions.fromMap(dataTable.options());
         SnapshotManager snapshotManager = dataTable.snapshotManager();
         Long snapshotId = coreOptions.scanSnapshotId();
+        String tagName = coreOptions.scanTagName();
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java:202]: Here add a check, if both snapshotId and tagName are not null, then there is ambiguity here.

Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java:202]: tagName had checked follow line：if (!StringUtils.isEmpty(tagName) && dataTable.tagManager().tagExists(tagName))

File: paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java (hunk 1 above)
Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java:175]: tagName had checked follow line：if (!StringUtils.isEmpty(tagName) && dataTable.tagManager().tagExists(tagName))，so here may not need add a check.

File: paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java (hunk 2 above)
Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java:215]: yes ，change to else

File: paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java (hunk 1 above)
Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java:175]: If user both set snapshot-id and tag-name would be ambiguity here  would add a warning log for it, if user both set, would use snapshot-id firstly.

**Comment by LinMingQiang** [comment]
+1

**Comment by wwj6591812** [comment]
+1

**Comment by wwj6591812** [comment]
+1

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**